Assets/MixedRealityToolkit-SDK/Features/Boundary/System/MixedRealityBoundarySystem.cs

```diff
@@ -0,0 +1,491 @@
+
+            if (FloorHeight.Value > location.y ||
+                BoundaryHeight < location.y)
+            {
+                // Location below the floor or above the boundary height.
+                return false;
+            }
+
+            // Boundary coordinates are always "on the floor"
+            Vector2 point = new Vector2(location.x, location.z);
+
+            if (boundaryType == UnityEngine.Experimental.XR.Boundary.Type.PlayArea)
+            {
+                // Check the inscribed rectangle.
+                if (rectangularBounds != null)
+                {
+                    return rectangularBounds.IsInsideBoundary(point);
+                }
+            }
+            else if (boundaryType == UnityEngine.Experimental.XR.Boundary.Type.TrackedArea)
+            {
+                // Check the geometry
+                return EdgeUtilities.IsInsideBoundary(Bounds, point);
+            }
+
+            // Not in either boundary type.
+            return false;
+        }
+
+        /// <inheritdoc/>
+        public bool TryGetRectangularBoundsParams(out Vector2 center, out float angle, out float width, out float height)
+        {
+            if (rectangularBounds == null || !rectangularBounds.IsValid)
+            {
+                center = EdgeUtilities.InvalidPoint;
+                angle = 0f;
+                width = 0f;
+                height = 0f;
+                return false;
+            }
+
+            // Handle the user teleporting (boundary moves with them).
+            Vector3 transformedCenter = CameraCache.Main.transform.parent.TransformPoint(
+                new Vector3(rectangularBounds.Center.x, 0f, rectangularBounds.Center.y));
+
+            center = new Vector2(transformedCenter.x, transformedCenter.z);
+            angle = rectangularBounds.Angle;
+            width = rectangularBounds.Width;
+            height = rectangularBounds.Height;
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public GameObject GetPlayAreaVisualization()
+        {
+            if (currentPlayArea != null)
+            {
+                return currentPlayArea;
+            }
+
+            // Get the rectangular bounds.
+            Vector2 center;
+            float angle;
+            float width;
+            float height;
+
+            if (!TryGetRectangularBoundsParams(out center, out angle, out width, out height))
+            {
+                // No rectangular bounds, therefore cannot create the play area.
+                return null;
+            }
+
+            // Render the rectangular bounds.
+            if (EdgeUtilities.IsValidPoint(center))
+            {
+                currentPlayArea = GameObject.CreatePrimitive(PrimitiveType.Quad);
+                currentPlayArea.name = "Boundary System Play Area";
+                currentPlayArea.transform.Translate(new Vector3(center.x, 0.005f, center.y)); // Add fudge factor to avoid z-fighting
```

Comment:
could we add this "fudge" factor to the profile settings?

Reply:
Just trying to get the "majick" numbers out 😉

Reply:
there's another way we can fix this... by adjusting the render order between this quad and the floor... let's file an issue for beta so we can determine the best way to address.

Reply:
Tracking the offset via #2552